fe/fe-core/src/main/java/com/starrocks/sql/optimizer/base/DistributionSpec.java

```diff
@@ -35,6 +43,43 @@ public boolean isSinglePartition() {
         public boolean isEmptyPartition() {
             return partitionIds.size() == 0;
         }
+
+        public void addJoinEquivalentPair(int leftColumn, int rightColumn) {
+            ColumnRefSet columns = null;
+            if (joinEquivalentColumns.containsKey(leftColumn)) {
+                columns = joinEquivalentColumns.get(leftColumn);
+            }
+            if (joinEquivalentColumns.containsKey(rightColumn)) {
+                columns = joinEquivalentColumns.get(rightColumn);
+            }
+            if (columns == null) {
+                columns = new ColumnRefSet();
+            }
+            columns.union(leftColumn);
+            columns.union(rightColumn);
+            joinEquivalentColumns.putIfAbsent(leftColumn, columns);
+            joinEquivalentColumns.putIfAbsent(rightColumn, columns);
```

Comment:
The logical looks is mistakes
A.a = B.b, C.c = D.d, A.a = C.c

The result wil be ?
```
A.a -> [A.a, B.b, C.c]
B.b -> [A.a, B.b]
C.c -> [C.c, D.d]
D.d -> [C.c, D.d]
```

Reply:
`A.a`、`B.b`、`C.c`、`D.d` share the same instance of `ColumnRefSet`

Reply:
I think `A.a, B.b` is an instance, `C.c, D.d` is another instance

Reply:
updated